db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java

```diff
@@ -526,6 +526,32 @@ public void bindExpressions(FromList fromListParam) throws StandardException{
             return;
         }
 
+        // Evaluate expressions with constant operands here to simplify the
+        // query tree and to reduce the runtime cost. Do it before optimize()
+        // since the simpler tree may have more accurate information for
+        // the optimizer. (Example: The selectivity for 1=1 is estimated to
+        // 0.1, whereas the actual selectivity is 1.0. In this step, 1=1 will
+        // be rewritten to TRUE, which is known by the optimizer to have
+        // selectivity 1.0.)
+        // This is also done before predicate simplification to enable
+        // more predicates to be pruned away.
+        accept(new ConstantExpressionVisitor());
+
+        // Perform predicate simplification.  Currently only
+        // simple rewrites involving boolean TRUE/FALSE are done, such as:
+        // TRUE AND col1 IN (1,2,3)  ==>  col1 IN (1,2,3)
+        // FALSE OR col1 = 1         ==>  col1 = 1
+        //
+        // Predicate simplification is done before binding because
+        // Subqueries and aggregates in the WHERE clause and HAVING
+        // clause get added to the whereSubquerys, whereAggregates,
+        // havingSubquerys and havingAggregates lists during binding.
+        // If the predicates containing thoses SubqueryNodes and
+        // AggregateNodes are subsequently removed from the WHERE or
+        // HAVING clause, query compilation will fail.
+        if (!getCompilerContext().getDisablePredicateSimplification())
```

Comment:
Should we restrict it to just WHERE and HAVING clause of a SELECT node?

Reply:
Yes, made the change.